 puma01_sim:
  # Publish all joint states ------------------------------------------
  sim_joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 200

# Trajectory controller -----------------------------------------------

  sim_trajectory_controller:
    type: "effort_controllers/JointTrajectoryController"

    joints:
      - joint_1
      - joint_2
      - joint_3
      - joint_4
      - joint_5
      - joint_6

    constraints:
      goal_time: 0.5                   
      stopped_velocity_tolerance: 0.02

      joint_1:
        trajectory: 0.05               
        goal: 0.02         

      joint_2:
        trajectory: 0.05               
        goal: 0.02     

      joint_3:
        trajectory: 0.05               
        goal: 0.02        

      joint_4:
        trajectory: 0.05               
        goal: 0.02         

      joint_5:
        trajectory: 0.05               
        goal: 0.02         

      joint_6:
        trajectory: 0.05               
        goal: 0.02      

    gains: 

    # for computed_torque_control implementation
      # joint_1: {p: 1.0, i: 0.0, d: 2.0} 
      # joint_2: {p: 30.0, i: 0.0, d: 12.0}
      # joint_3: {p: 40.0, i: 0.0, d: 35.0}
      # joint_4: {p: 1.0, i: 0.0, d: 0.5} 
      # joint_5: {p: 1.0, i: 0.0, d: 0.5}
      # joint_6: {p: 1.0, i: 0.0, d: 0.5}

    # for simple PID
      joint_1: {p: 100.0, i: 0.0, d: 2.0} 
      joint_2: {p: 400.0, i: 0.0, d: 12.0}
      joint_3: {p: 400.0, i: 0.0, d: 35.0}
      joint_4: {p: 5.0, i: 0.0, d: 0.5} 
      joint_5: {p: 5.0, i: 0.0, d: 0.5}
      joint_6: {p: 5.0, i: 0.0, d: 0.5}

    state_publish_rate:  25            
    action_monitor_rate: 30           
    stop_trajectory_duration: 0     

  # Computed torque controller -------------------------------

  sim_computed_torque_controller:
    type: "puma01_controllers/ComputedTorqueController"
    joints:
      - joint_1
      - joint_2
      - joint_3
      - joint_4
      - joint_5
      - joint_6
    joint_1: 
      pid: {p: 30.0, i: 0.0, d: 5.0}
    joint_2: 
      pid: {p: 30.0, i: 0.0, d: 10.0}
    joint_3: 
      pid: {p: 40.0, i: 0.0, d: 27.0}
    joint_4: 
      pid: {p: 10.0, i: 0.0, d: 0.5}
    joint_5: 
      pid: {p: 10.0, i: 0.0, d: 0.5}
    joint_6: 
      pid: {p: 10.0, i: 0.0, d: 0.5}


  # Effort Controllers --------------------------------------------

  joint_1_effort_controller:
    type: effort_controllers/JointEffortController
    joint: joint_1
    pid: {p: 30.0, i: 0.0, d: 10.0}

  joint_2_effort_controller:
    type: effort_controllers/JointEffortController
    joint: joint_2
    pid: {p: 30.0, i: 0.0, d: 10.0}

  joint_3_effort_controller:
    type: effort_controllers/JointEffortController
    joint: joint_3
    pid: {p: 40.0, i: 0.0, d: 27.0}

  joint_4_effort_controller:
    type: effort_controllers/JointEffortController
    joint: joint_4
    pid: {p: 10.0, i: 0.0, d: 0.5}

  joint_5_effort_controller:
    type: effort_controllers/JointEffortController
    joint: joint_5
    pid: {p: 10.0, i: 0.0, d: 0.5}

  joint_6_effort_controller:
    type: effort_controllers/JointEffortController
    joint: joint_6
    pid: {p: 10.0, i: 0.0, d: 0.5}


  # Position Controllers ---------------------------------------

  # joint_1_controller:
  #   type: effort_controllers/JointPositionController
  #   joint: joint_1
  #   pid: {p: 30.0, i: 0.0, d: 25.0}

  # joint_2_controller:
  #   type: effort_controllers/JointPositionController
  #   joint: joint_2
  #   pid: {p: 30.0, i: 0.0, d: 25.0}

  # joint_3_controller:
  #   type: effort_controllers/JointPositionController
  #   joint: joint_3
  #   pid: {p: 30.0, i: 0.0, d: 25.0}

  # joint_4_controller:
  #   type: effort_controllers/JointPositionController
  #   joint: joint_4
  #   pid: {p: 10.0, i: 0.0, d: 10.0}

  # joint_5_controller:
  #   type: effort_controllers/JointPositionController
  #   joint: joint_5
  #   pid: {p: 10.0, i: 0.0, d: 10.0}

  # joint_6_controller:
  #   type: effort_controllers/JointPositionController
  #   joint: joint_6
  #   pid: {p: 10.0, i: 0.0, d: 10.0}
